Scroll fork protection

jeteve opened the pull request:
This implements issue #79 by doing two things:

- Avoid finishing a Scroll helper when the finishing process is not the one who created it.
- Crash with an Illegal exception when a programmer attempts to use the Async helper in a forked process.

jeteve commented on .gitignore:
```diff
@@ -11,3 +11,5 @@ log
 *.xml
 shield/*
+# App::CSE
+.cse*
```
This is https://metacpan.org/pod/App::CSE 's index.

jeteve commented:
@clintongormley Any thoughts on that? Cheers.

clintongormley commented:
Hi @jeteve 

Thanks for the PR.  Sorry for that late reply, been busy getting Elasticsearch 2.0.0-beta1 out :)  (w00t)

Please could I ask you to sign the CLA? https://www.elastic.co/contributor-agreement

thanks

jeteve commented:
@clintongormley done:)

jeteve commented:
Hi @clintongormley , Any news on that one?

clintongormley commented:
Hi @jeteve 

Thanks a lot for the PR.  I reworked the code a bit so that it wasn't checking the PID on every call
to next (which strictly speaking is allowed in a child process) but only on attempts to refill the buffer.
I extended it (and the tests) to work with Search::Elasticsearch::Async::Scroll, which highlighted some
issues with the finished() method, which I fixed as well.

many thanks for the work

jeteve commented:
Hi Clinton,

Thanks a lot for that! Looking forward to installing the new version :)

Jerome.

On 6 September 2015 at 19:10, Clinton Gormley <notifications@github.com>
wrote:

> Hi @jeteve <https://github.com/jeteve>
>
> Thanks a lot for the PR. I reworked the code a bit so that it wasn't
> checking the PID on every call
> to next (which strictly speaking is allowed in a child process) but only
> on attempts to refill the buffer.
> I extended it (and the tests) to work with
> Search::Elasticsearch::Async::Scroll, which highlighted some
> issues with the finished() method, which I fixed as well.
>
> many thanks for the work
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch-perl/pull/80#issuecomment-138104831>
> .
>



-- 
Jerome Eteve
+44(0)7738864546
http://www.eteve.net/
